example/output/json/components.callbacks/userCreatedCallback.json

```diff
@@ -0,0 +1,28 @@
+{
```

Comment:
does storing these kind of files make sense?
They are not a valid json schema, so how can we use them?

Reply:
I changed `oas2tson` only, so any issues with other commands are existing issues not related to this PR or the issue that motivated it.

I don't know how reliable they are, but I found a couple of online JSON schema validators and both say it is valid JSON Schema by drafts 4 through 7.

Maybe I misunderstood https://github.com/nearform/openapi-transformer-toolkit/issues/305#issuecomment-2283419148. I thought this comment meant `-p` should everything. If you had a narrower scope in mind, I can adjust it I know what it is.

My thoughts on the topic for `oas2tson`...

`oas2tson` supports Fastify [validation and serialization](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/#validation) `json-schema-to-ts` type provider per https://github.com/nearform/openapi-transformer-toolkit/issues/35 . Given that intent and what the OpenAPI spec defines:

- `components.schemas` -- makes sense because we can use them to compose [`RouteOptions`](https://fastify.dev/docs/latest/Reference/Routes/#routes-options)
- `components.requestBodies` -- probably won't work for a `body` validation schema, but could work if you select the `x.content.['application/json`]` member, maybe? ([OpenAPI](https://spec.openapis.org/oas/latest.html#request-body-object); note the `content` field)
- `components.headers` -- don't match what Fastify expects for a header validation schema ([OpenAPI](https://spec.openapis.org/oas/latest.html#header-object))
- `components.parameters` -- don't match what Fastify expects for `params`, `querystring`, etc. ([OpenAPI](https://spec.openapis.org/oas/latest.html#header-object))
- `components.examples` -- aren't used by Fastify directly
- `components.links` -- aren't used by Fastify directly
- `components.callbacks` -- aren't used by Fastify directly
- `webhooks` -- aren't used by Fastify as far as I can tell; not my area of expertise, so may be missing something
- `paths` -- cannot be used by Fastify as `RouteOptions`

Which leads me to think supporting anything by `components.schemas` in `oas2tson` may not add value. For `oas2tson` users, the best approach is likely define schemas for everything, then define parameters in terms of schemas as below.

```yaml
components:
  schemas:
    PetQuerySchema:
      type: object
     properties:
     ...
  parameters:
    PetQuery:
      in: query
      schema:
        $ref: '#/components/schemas/PetQuerySchema'
     ...
```

Then use the TSON schemas:

```typescript
const routeSchema = {
  querystring: PetQuerySchema // from oas2tson ouput PetQuerySchema.ts
  ...
}
```

For other commands, supporting more parts of the OpenAPI spec may make sense. I don't know, but note that `oas2ts` and `json2ts` do not support `-p` and only export `components.schemas`. 

Any problems in other commands' output are existing problems and should be handled in separate issues -- and are likely due to the fact that `openapi-schema-to-json-schema` supports only Schema Object and Parameter Object. That may feed back into the original question about -p.